src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -717,155 +439,319 @@ private static OutputStream getOutputStream(String outputLocation) throws IOExce
+        @Option(names = {"-T", "--treeWithComments"},
+                description = "Print Abstract Syntax Tree(AST) of the file including comments")
+        private boolean printAstWithComments;
+
+        /** Option that controls whether to print the parse tree of the javadoc comment. */
+        @Option(names = {"-j", "--javadocTree"},
+                description = "Print Parse tree of the Javadoc comment")
+        private boolean printJavadocTree;
+
+        /** Option that controls whether to print the full AST of the file. */
+        @Option(names = {"-J", "--treeWithJavadoc"},
+                description = "Print full Abstract Syntax Tree of the file")
+        private boolean printTreeWithJavadoc;
+
+        /** Option that controls whether to print debug info. */
+        @Option(names = {"-d", "--debug"},
+                description = "Print all debug logging of CheckStyle utility")
+        private boolean debug;
+
+        /** Option that allows users to specify a list of paths to exclude. */
+        @Option(names = {"-e", "--exclude"},
+                description = "Directory path to exclude from CheckStyle")
+        private List<File> exclude = new ArrayList<>();
+
+        /** Option that allows users to specify a regex of paths to exclude. */
+        @Option(names = {"-x", "--exclude-regexp"},
+                description = "Regular expression of directory to exclude from CheckStyle")
+        private List<Pattern> excludeRegex = new ArrayList<>();
+
+        /** Switch whether to execute ignored modules or not. */
+        @Option(names = "--executeIgnoredModules",
+                description = "Allows ignored modules to be run.")
+        private boolean executeIgnoredModules;
+
+        /** The checker threads number. */
+        @Option(names = {"-C", "--checker-threads-number"}, description = "(experimental) The "
+                + "number of Checker threads (must be greater than zero)")
+        private int checkerThreadsNumber = DEFAULT_THREAD_COUNT;
 
+        /** The tree walker threads number. */
+        @Option(names = {"-W", "--tree-walker-threads-number"}, description = "(experimental) The "
+                + "number of TreeWalker threads (must be greater than zero)")
+        private int treeWalkerThreadsNumber = DEFAULT_THREAD_COUNT;
+
+        /**
+         * Gets the list of exclusions provided through the command line arguments.
+         * @return List of exclusion patterns.
+         */
+        private List<Pattern> getExclusions() {
+            final List<Pattern> result = new ArrayList<>();
+            exclude.forEach(file -> result.add(
+                    Pattern.compile("^" + Pattern.quote(file.getAbsolutePath()) + "$")));
+            result.addAll(excludeRegex);
+            return result;
+        }
+
+        /**
+         * Determines the files to process.
+         * @return list of files to process
+         */
+        private List<File> getFilesToProcess() {
```

Comment:
can this , and few method below be returned to Main ?

Reply:
@remkop ,
I might miss smth , but , it is use one place at https://github.com/checkstyle/checkstyle/blob/ad9d8ec298752a589597c3aae58a30da253fad48/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L169 that in Main class. 
It will require cli.getExclusions() but we can change method and put report of this method as argument.

Reply:
Ok, let me take another look.